sdk/eventhub/Azure.Messaging.EventHubs/src/PartitionContext.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs
+{
+    /// <summary>
+    ///   Contains information about a partition, like its identifier and information about its last
+    ///   enqueued event.
+    /// </summary>
+    ///
+    public class PartitionContext
```

Comment:
I think this makes more sense as a struct, if you wouldn't mind changing it.  That's going to require a public constructor, which is a reasonable trade-off.